### ISSUE TITLE
The Azure App Insights key is not a valid GUID

### ISSUE THREAD

**Issue by allengeorgek** [OfficeDev/microsoft-teams-apps-company-communicator#977]
![image](https://user-images.githubusercontent.com/22046702/222070004-aefeb615-06ce-47b3-a877-6969482740a3.png)

Tried to deploy the app for a customer today & received the following error similar to the ones being discussed.
Generated new GUID & still the same error message while validating the Azure App Insights Key. Says it is not a valid GUID.

Raw Error:
{  "code": "InvalidTemplateDeployment",  "details": [    {      "details": [        {          "code": "InvalidBotData",          "target": "TeamsCommunicator-author",          "message": "The Azure App Insights key is not a valid GUID"        },        {          "code": "InvalidBotData",          "target": "TeamsCommunicator",          "message": "The Azure App Insights key is not a valid GUID"        }      ]    }  ],  "message": "The template deployment 'Microsoft.Template-20230301122158' is not valid according to the validation procedure. The tracking id is '178caf13-3f5b-47d6-874a-c0d5eda0d90f'. See inner errors for details."}

**Comment by gsv022** [comment]
Hi @allengeorgek ,
Thanks for raising the issue. We are working on this issue already. Duplicate to [975) (https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/975)

**Comment by allengeorgek** [comment]
hi @gsv022 - Do we have any ETA on this? Customer CXO team is actively involved in the deployment of this app & hence very critical to have this resolved soon.

**Comment by v-royavinash** [comment]
@allengeorgek , We are prioritizing this issue as our highest concern, and we have tentatively identified the problem. Currently, we are testing a potential solution, and we will confirm once we have found a working fix.

**Comment by v-jaygupta** [comment]
Hi @allengeorgek,
We identified that the issue is not specific to the company communicator but is due to a recent update in either Bot Services or Azure template custom validation. The update is preventing the creation of a reference of App Insights Instrument key to the Microsoft Bot Service configuration. This change is impacting other templates and repositories that are dependent on it, including company communicator. 
We have identified a possible workaround and are currently testing it with the highest priority. As soon as testing is complete, we will share the steps to unblock the deployment. Thank you for your patience and understanding as we work to resolve this issue.

**Comment by gsv022** [comment]
Hi @allengeorgek ,

- It is identified that the issue is not specific to the company communicator but is evident that there is an issue with Azure ARM resources functionality.
- We have updated the [Deploy to Azure](https://portal.azure.com/?feature.customportal=false/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FOfficeDev%2Fmicrosoft-teams-company-communicator-app%2Fmain%2FDeployment%2Fazuredeploy.json) link in 2nd step of the CC Deployment Guide. The earlier faced issue "_The Azure App Insights key is not a valid GUID_" is now resolved and deployment is working as expected.
- Please try deploying the Company communicator app and let us know if you face any further issues. 
- If the deployment is failing due to role assignments, please refer to the [Troubleshooting Guide](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/wiki/Troubleshooting#4-multiple-instance-of-company-communicator-in-the-same-tenant-action-items).

**Comment by gsv022** [comment]
@allengeorgek , Please let us know the status if the issue persists. Otherwise we are good to close out the issue.

**Comment by gsv022** [comment]
@allengeorgek, We are closing this request as we have provided the inputs for the query raised. If you face any issue with company communicator please raise a new issue for further assistance.
